app/src/main/java/org/p2p/wallet/push_notifications/repository/PushTokenRepository.kt

```diff
@@ -0,0 +1,28 @@
+package org.p2p.wallet.push_notifications.repository
+
+import com.google.firebase.messaging.FirebaseMessaging
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+class PushToken(val value: String)
+
+class PushTokenRepository {
+
+    private val firebaseMessaging: FirebaseMessaging
+        get() = FirebaseMessaging.getInstance()
+
+    suspend fun getPushToken(): PushToken = suspendCancellableCoroutine { continuation ->
+        firebaseMessaging.token
+            .addOnSuccessListener { continuation.resume(PushToken(it)) }
+            .addOnCanceledListener { continuation.cancel() }
+            .addOnFailureListener { continuation.resumeWithException(it) }
+    }
+
+    suspend fun deletePushToken(): Unit = suspendCancellableCoroutine { continuation ->
+        firebaseMessaging.deleteToken()
```

Comment:
why should we even deleteToken at the first place?

Reply:
It removes push token from FCM to prevent this device from receiving notifications

Reply:
isn't this is a Notification Server responsibility? Like we send a request to NotificationServer to remove token, so it wont send push notifications

Reply:
It's a good point, it should be server responsibility